configTools/src/test/scala/com/gu/janus/policy/IamSpec.scala

```diff
@@ -0,0 +1,269 @@
+package com.gu.janus.policy
+
+import com.gu.janus.policy.Iam._
+import io.circe.JsonObject
+import io.circe.parser.decode
+import io.circe.syntax._
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+
+class IamSpec extends AnyFreeSpec with Matchers {
+
+  "Action" - {
+    "should encode action name correctly" in {
+      Action("s3:GetObject").asJson.noSpaces shouldBe """"s3:GetObject""""
+    }
+  }
+
+  "Resource" - {
+    "should encode resource id correctly" in {
+      Resource(
+        "arn:aws:s3:::my-bucket/*"
+      ).asJson.noSpaces shouldBe """"arn:aws:s3:::my-bucket/*""""
+    }
+  }
+
+  "Condition" - {
+    "should encode single condition correctly" in {
+      val condition = Condition(
+        key = "aws:SourceIp",
+        typeName = "StringEquals",
+        conditionValues = List("203.0.113.0/24")
+      )
+
+      val expected = """{"StringEquals":{"aws:SourceIp":["203.0.113.0/24"]}}"""
+      condition.asJson.noSpaces shouldBe expected
+    }
+
+    "should encode multiple condition values correctly" in {
+      val condition = Condition(
+        key = "aws:SourceIp",
+        typeName = "StringEquals",
+        conditionValues = List("203.0.113.0/24", "203.0.113.1/24")
+      )
+
+      val expected =
+        """{"StringEquals":{"aws:SourceIp":["203.0.113.0/24","203.0.113.1/24"]}}"""
+      condition.asJson.noSpaces shouldBe expected
+    }
+  }
+
+  "Principal" - {
+    "should encode single principal correctly" in {
+      val principal = Principal("AROAXXXXXXXXXXXXXXX", "AWS")
+      val expected = """{"AWS":"AROAXXXXXXXXXXXXXXX"}"""
+      principal.asJson.noSpaces shouldBe expected
+    }
+  }
+
+  "Statement" - {
+    val basicStatement = Statement(
+      effect = Effect.Allow,
+      actions = List(Action("iam:PutRolePolicy"), Action("iam:getRole")),
+      resources = List(Resource("*")),
+      id = None
+    )
+
+    "should encode basic statement correctly" in {
+      val expected = """{
+                         |"Effect":"Allow",
+                         |"Action":["iam:PutRolePolicy","iam:getRole"],
+                         |"Resource":["*"]
+                         |}""".stripMargin.replaceAll("\\s", "")
+
+      basicStatement.asJson.noSpaces shouldBe expected
+    }
+
+    "should encode statement with ID correctly" in {
+      val statementWithId = basicStatement.copy(id = Some("statement1"))
+      val expected = """{
+                         |"Effect":"Allow",
+                         |"Action":["iam:PutRolePolicy","iam:getRole"],
+                         |"Resource":["*"],
+                         |"Sid":"statement1"
+                         |}""".stripMargin.replaceAll("\\s", "")
+
+      statementWithId.asJson.noSpaces shouldBe expected
```

Comment:
Would it help debugging test failures for all these JSON tests if we compare a pretty printed version? Something like `spaces2` in place of `noSpaces` would make it easier to visually parse the diffs that this test case genera, and would mean the expected value above could be formatted to match - without the need for the `replaceAll` call.

Reply:
Done in 7108fccfe6e96ad3753ae781cc371b2f6e09f07c